### PR TITLE
Preliminary support for Ubuntu 21.04

### LINKED ISSUE #3443: [BUG] Dependency missing on Debian Bullseye (11)
The renamed library in Debian 11 from lib32gcc1 to lib32gcc-s1.

The issue is fixed in [this](https://github.com/GameServerManagers/LinuxGSM/pull/2951/commits/2ab3ed0d3eb29b1b22b23609c3a38ba8c20f3d86) commit but the added check fails because the testing version of Bullseye returns no version number. 

Suggested fix is to parse the output of lsb_release -d that contains the release name (bullseye) instead of trying to regex a number out.

Lazy and very wrong workaround for the moment is to modify the line 356 of check_deps.sh to:
`array_deps_required+=( lib32gcc-s1 lib32stdc++6 )`

### PR DESCRIPTION
**Pull request by olokos**
Seems like without this the steamcmd will never run as steamcmdfail is set to 1 despite having neccesary requirements.

# Description

Check deps doesn't check for 21.04 ubuntu, now it does.

Fixes #3443 but only for ubuntu 21.04 users

## Type of change

* [x] Bug fix (a change which fixes an issue).

## Checklist

PR will not be merged until all steps are complete.

* [x] This pull request links to an issue.
* [x] This pull request uses the `develop` branch as its base.
* [x] This pull request Subject follows the Conventional Commits standard.
* [x] This code follows the style guidelines of this project.
* [x] I have performed a self-review of my code.
* [x] I have checked that this code is commented where required.
* [x] I have provided a detailed with enough description of this PR.
* [x] I have checked If documentation needs updating.

## Documentation

If documentation does need updating either update it by creating a PR (preferred) or request a documentation update.
* User docs: https://github.com/GameServerManagers/LinuxGSM-Docs
* Dev docs: https://github.com/GameServerManagers/LinuxGSM-Dev-Docs

**Thank you for your Pull Request!**

On top of that there's this issue:
https://media.discordapp.net/attachments/424152809970204672/837743019247534170/unknown.png?width=720&height=324

Which I resolved by doing
sudo apt install gcc-multilib

and then 
cp /lib/x86_64-linux-gnu/libgcc_s.so.1 /home/csgoserver/libgcc_s.so.1

Once new libgcc_s.so.1 is provided to lgsm, the server works again.

### CODE REVIEW

**Comment by olokos** [comment]
```
rm /home/csgoserver/csgoserver-1/serverfiles/bin/libgcc_s.so.1
ln -s /lib/x86_64-linux-gnu/libgcc_s.so.1 /home/csgoserver/csgoserver-1/serverfiles/bin/libgcc_s.so.1
```

this is what needs to be done also on Ubuntu 21.04

File: lgsm/functions/check_deps.sh [hunk 1]
```diff
@@ -350,7 +350,7 @@ fn_deps_build_debian(){
 	if [ "${shortname}" != "ts3" ]&&[ "${shortname}" != "mumble" ]&&[ "${shortname}" != "mc" ]&&[ "${shortname}" != "pmc" ]&&[ "${shortname}" != "wmc" ]&&[ "${engine}" != "renderware" ]; then
 		if [ "${arch}" == "x86_64" ]; then
 			# lib32gcc1 is now called lib32gcc-s1 in debian 11
```
Comment (guidorugo) [inline lgsm/functions/check_deps.sh:352]: I think is a good idea to add the rename for ubuntu as well in the comment

**Comment by Sidicer** [comment]
cp /lib/x86_64-linux-gnu/libgcc_s.so.1 /home/sidicer/serverfiles/libgcc_s.so.1

^ this only helps if you had a server previously and you updated your distro to 21.*
Cannot install a new server or it's files due to dependencies always failing for the same reason "Package lib32gcc1 is not available, but is referred to by another package."

Manually putting libgcc_s.so.1 to "serverfiles" directory does not skip the dependencie check.

Impossible to setup any server for a new installation on OS: Ubuntu 21.04 x86_64

**Comment by guidorugo** [comment]
> cp /lib/x86_64-linux-gnu/libgcc_s.so.1 /home/sidicer/serverfiles/libgcc_s.so.1
> 
> ^ this only helps if you had a server previously and you updated your distro to 21.*
> Cannot install a new server or it's files due to dependencies always failing for the same reason "Package lib32gcc1 is not available, but is referred to by another package."
> 
> Manually putting libgcc_s.so.1 to "serverfiles" directory does not skip the dependencie check.
> 
> Impossible to setup any server for a new installation on OS: Ubuntu 21.04 x86_64

Did you try with the patch version instead of the cp ?

**Comment by Sidicer** [comment]
> > cp /lib/x86_64-linux-gnu/libgcc_s.so.1 /home/sidicer/serverfiles/libgcc_s.so.1
> > ^ this only helps if you had a server previously and you updated your distro to 21.*
> > Cannot install a new server or it's files due to dependencies always failing for the same reason "Package lib32gcc1 is not available, but is referred to by another package."
> > Manually putting libgcc_s.so.1 to "serverfiles" directory does not skip the dependencie check.
> > Impossible to setup any server for a new installation on OS: Ubuntu 21.04 x86_64
> 
> Did you try with the patch version instead of the cp ?

The command works, I meant that putting libgcc_s.so.1 into the serverfiles does not help.
Over-come this issue on ubuntu 21 by starting LinuxGSM, downloading its files, then editing the check_deps.sh and changed
`356: array_deps_required+=( lib32gcc1 lib32stdc++6 ) to array_deps_required+=( lib32gcc-s1 lib32stdc++6 )`

**Comment by olokos** [comment]
@Sidicer 
You need to check the main part of this Pull Request, which is the actual code changes I have implemented, in order to properly fix this problem.

The problem you're having is fixed by this PR completely, you can check the corrected file by clicking on the top "changed files" to see the modifications I've made to fix it.

That would resolve the problem you're having.

After you fix that problem though, you will have another one which I linked in the discord link.

the apt install and cp is a solution to the second problem, the one you would have once you manage to install the server.

Because you didn't use my fixed version of check_deps.sh, you didn't even get to the point where `apt install` and `cp` commands would be useful, as the problem with ubuntu 21.04 is twofold.

1. Check_deps.sh not considering 21.04
2. LGSM not using the latest version of gcc-multilib provided by the system.


If you haven't solved problem 1, with the changes to check_deps.sh I have made, you won't be able to reach problem 2. - especially with a fresh server installation.

**Comment by christianlux** [comment]
When will this be fixed in server scripts?

**Comment by Claiyc** [comment]
@christianlux dependency-checks are currently being refactored. Work is in progress but no ETA afaik.

**Comment by dgibbs64** [comment]
Full refactor of dependency checking is now required due to recent additions of Rocky & Alma Linux but dependency differences between Ubuntu and Debian releases. This will be addressed as a priority refactor but please bear with me as I am moving house soon.

**Pull request merged**